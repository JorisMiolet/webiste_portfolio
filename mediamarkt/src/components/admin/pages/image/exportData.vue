<script>
import imageData from '../dummyData.json';
import {PCImage} from "@/image.js";
import axios from "axios";
import * as XLSX from "xlsx";
export default {
  name: 'exportData',
  components: {},
  inject: ['url'],
  props: {
    isAdmin: Boolean,
  },
  data() {
    return {
      images: [],
      dummyData: imageData,
      selectedImage: null,
      searchFilter: null,
      summary: [],
      currentPage: 1,
      rowsPerPage: 10,
    }
  },
  created() {
    this.loadAllImages();
    this.loadImageSummary();
    console.log(this.images)
  },
  methods: {
    createOffer(image) {
      image = PCImage.createSampleImage(
          image.ArticleNR,
          image.EAN,
          image.Brand,
          image.Description,
          image.PROCESSOR,
          image.RAM,
          image.STORAGE,
          image.GPU,
      )
      this.images.push(image);
    },
    editImage(pcimage) {
      this.selectedImage = pcimage;
      this.toggleView();
    },
    deleteImage(image) {
      const confirmDelete = confirm(`are you sure you want to delete image ${image["Article NR"]}`);
      if (confirmDelete) {
        const urlWithQuery = `${this.url}/api/images/${image["id"]}`;
        axios.delete(urlWithQuery)
            .then(this.loadAllImages);
      }
    },
    resetImage() {
      this.$router.push(this.$route.matched[0].path)
    },
    onSelect(image) {
      this.selectedImage = image;
    },
    filterImages(){
      const urlWithQuery = `${this.url}/api/images/search?Filter=${this.searchFilter}`;
      axios.get(urlWithQuery)
          .then(response => this.images = response.data)
          .then(console.log(this.images))
    },
    loadAllImages() {
      const urlWithQuery = `${this.url}/api/images/all`;
      axios.get(urlWithQuery)
          .then(response => this.images = response.data)
          .then(console.log(this.images))
    },
    loadCompletedImages(){
      const urlWithQuery = `${this.url}/api/images/completed`;
      axios.get(urlWithQuery)
          .then(response => this.images = response.data)
          .then(console.log(this.images))
    },
    loadInCompletedImages(){
      const urlWithQuery = `${this.url}/api/images/incomplete`;
      axios.get(urlWithQuery)
          .then(response => this.images = response.data)
          .then(console.log(this.images))
    },
    loadImageSummary(){
      const urlWithQuery = `${this.url}/api/images/summary`;
      axios.get(urlWithQuery)
          .then(response => this.summary = response.data)
    },
    loadOutdatedImages(){
      const urlWithQuery = `${this.url}/api/images/outdated`;
      axios.get(urlWithQuery)
          .then(response => this.images = response.data)
          .then(console.log(this.images))
    },
    exportToExcel() {
      const worksheet = XLSX.utils.json_to_sheet(this.images);
      const workbook = XLSX.utils.book_new();
      XLSX.utils.book_append_sheet(workbook, worksheet, 'Sheet1');
      const excelBuffer = XLSX.write(workbook, {bookType: 'xlsx', type: 'array'});

      const blob = new Blob([excelBuffer], {type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'});
      const url = URL.createObjectURL(blob);

      const a = document.createElement('a');
      a.href = url;
      a.download = 'exported_data.xlsx';
      a.click();
      URL.revokeObjectURL(url);
    },
    prevPage() {
      if (this.currentPage > 1) {
        this.currentPage--;
      }
    },
    nextPage() {
      if (this.currentPage < this.totalPages) {
        this.currentPage++;
      }
    },
    pickup(image){
      const urlWithQuery = `${this.url}/api/images/pickup`;
      const data = {
        "article_nr": image["Article NR"],
        "user_id": sessionStorage.getItem('user_id')
      }
      axios.post(urlWithQuery, data)
          .then(this.loadAllImages())


    },
    getUserEmail(user) {
      return user ? user.email : 'N/A'; // Modify this based on your user object structure
    },
  },
  computed: {
    totalPages(){
      return Math.ceil(this.images.length / this.rowsPerPage)
    },
    paginatedImages() {
      const startIndex = (this.currentPage - 1) * this.rowsPerPage;
      const endIndex = startIndex + this.rowsPerPage;
      return this.images.slice(startIndex, endIndex);
    },

  },
  watch: {
    searchFilter() {
      this.filterImages();
    },
  },

}

</script>

<template>
  <div class="p-6">
    <div class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 mb-6 mt-40 md:mt-0">
      <div class="bg-white rounded-md border border-gray-100 p-6 shadow-md shadow-black/5">
        <div class="flex justify-between mb-6">
          <div>
            <div v-if="this.summary.completed" class="text-2xl font-semibold mb-1">{{ this.summary.completed }}</div>
            <div v-else class="text-2xl font-semibold mb-1">0</div>
            <div class="text-sm font-medium text-gray-600">Completed</div>
          </div>
        </div>
      </div>
      <div class="bg-white rounded-md border border-gray-100 p-6 shadow-md shadow-black/5">
        <div class="flex justify-between mb-6">
          <div>
            <div v-if="this.summary.in_progress" class="text-2xl font-semibold mb-1">{{ this.summary.in_progress }}</div>
            <div v-else class="text-2xl font-semibold mb-1">0</div>
            <div class="text-sm font-medium text-gray-600">In progress</div>
          </div>
        </div>
      </div>
      <div class="bg-white rounded-md border border-gray-100 p-6 shadow-md shadow-black/5">
        <div class="flex justify-between mb-6">
          <div>
            <div v-if="this.summary.outdated" class="text-2xl font-semibold mb-1">{{ this.summary.outdated }}</div>
            <div v-else class="text-2xl font-semibold mb-1">0</div>
            <div class="text-sm font-medium text-gray-600">Outdated</div>
          </div>
        </div>
      </div>
    </div>
    <div class="mb-6">
      <div class="bg-white border border-gray-100 shadow-md shadow-black/5 p-6 rounded-md">
        <div class="flex justify-center sm:justify-between mb-4 sm:items-start">
          <div class="font-medium ">Manage images</div>
        </div>
        <div class="flex flex-col sm:flex-row items-center mb-4 order-tab">
          <button @click="loadAllImages" type="button" class="bg-gray-50 text-sm font-medium text-gray-400 py-2 px-4 rounded-tl-md rounded-bl-md hover:text-gray-600 active">All</button>
          <button @click="loadCompletedImages" type="button" class="bg-gray-50 text-sm font-medium text-gray-400 py-2 px-4 rounded-tl-md rounded-bl-md hover:text-gray-600 active">Completed</button>
          <button @click="loadInCompletedImages" type="button" class="bg-gray-50 text-sm font-medium text-gray-400 py-2 px-4 hover:text-gray-600">In progress</button>
          <button @click="loadOutdatedImages" type="button" class="bg-gray-50 text-sm font-medium text-gray-400 py-2 px-4 rounded-tr-md rounded-br-md hover:text-gray-600">Outdated</button>
          <button type="button" class="sm:ml-auto bg-red-500 text-sm font-medium text-white py-2 px-4 rounded-tr-md rounded hover:bg-red-600"><router-link :to="{name: 'createImage'}">Create image</router-link></button>
        </div>
        <div class="flex items-center mb-4">
          <div class="relative w-full mr-2">
            <input v-model="searchFilter" type="text" class="py-2 pr-4 pl-10 bg-gray-50 w-full outline-none border border-gray-100 rounded-md text-sm focus:border-blue-500" placeholder="Search...">
            <i class="ri-search-line absolute top-1/2 left-4 -translate-y-1/2 text-gray-400"></i>
          </div>
        </div>
        <div class="overflow-x-auto">
          <table class="w-full min-w-[540px]" data-tab-for="order" data-page="active">
            <thead>
            <tr>
              <th class="text-[12px] uppercase tracking-wide font-medium text-gray-400 py-2 px-4 bg-gray-50 text-left rounded-tl-md rounded-bl-md">Article number</th>
              <th class="text-[12px] uppercase tracking-wide font-medium text-gray-400 py-2 px-4 bg-gray-50 text-left">EAN</th>
              <th class="text-[12px] uppercase tracking-wide font-medium text-gray-400 py-2 px-4 bg-gray-50 text-left">Brand</th>
              <th class="text-[12px] uppercase tracking-wide font-medium text-gray-400 py-2 px-4 bg-gray-50 text-left">Status</th>
              <th class="text-[12px] uppercase tracking-wide font-medium text-gray-400 py-2 px-4 bg-gray-50 text-left rounded-tr-md rounded-br-md">User</th>
              <th class="text-[12px] uppercase tracking-wide font-medium text-gray-400 py-2 px-4 bg-gray-50 text-left rounded-tr-md rounded-br-md">Actions</th>
            </tr>
            </thead>
            <tbody>
            <tr v-for="(pcimage) in paginatedImages" :key="pcimage.EAN">
              <td class="py-2 px-4 border-b border-b-gray-50">
                <span class="text-[13px] font-medium text-gray-800">{{ pcimage["Article NR"] }}</span>
              </td>
              <td class="py-2 px-4 border-b border-b-gray-50">
                <span class="text-[13px] font-medium text-gray-800">{{ pcimage["EAN"] }}</span>
              </td>
              <td class="py-2 px-4 border-b border-b-gray-50">
                <span class="text-[13px] font-medium text-gray-800">{{ pcimage["Brand"] }}</span>
              </td>
              <td class="py-2 px-4 border-b border-b-gray-50">
                <span class="text-[13px] font-medium text-gray-800">{{ pcimage["STATUS"] }}</span>
              </td>
              <td class="py-2 px-4 border-b border-b-gray-50">
                <span class="text-[13px] font-medium text-gray-800">{{ getUserEmail(pcimage["user"]) }}</span>
              </td>
              <td class="py-2 px-4 border-b border-b-gray-50">
                <span v-if="isAdmin" class="inline-block p-1 rounded bg-emerald-500/10 text-emerald-500 font-medium text-[12px] leading-none">
                  <button>
                    <router-link :to="{name:'editImage', params: {ArticleNR: pcimage['Article NR']}}" @click="onSelect(pcimage)">edit</router-link>
                  </button>
                </span>
                <span @click="deleteImage(pcimage)" v-if="isAdmin" class="inline-block ml-2 p-1 rounded bg-red-600/10 text-red-600 font-medium text-[12px] leading-none">
                  <button>
                    <span>delete</span>
                  </button>
                </span>
                <span @click="pickup(pcimage)" v-if="pcimage['STATUS'] !== 'completed' && pcimage['STATUS'] !== 'in progress'" class="inline-block ml-2 p-1 rounded bg-orange-500/10 text-orange-500 font-medium text-[12px] leading-none">
                  <button>
                    <span>pick up</span>
                  </button>
                </span>
              </td>
            </tr>
            </tbody>
          </table>
          <div class="flex justify-center mt-4">
            <button @click="prevPage" :disabled="currentPage === 1" :class="{'disabled': currentPage === 1}" class="bg-red-500 text-sm font-medium text-white py-2 px-4 rounded hover:bg-red-600">
              Previous
            </button>
            <div class="mx-4">
              Page {{ currentPage }} of {{ totalPages }}
            </div>
            <button @click="nextPage" :disabled="currentPage === totalPages" :class="{'disabled': currentPage === totalPages}" class="bg-red-500 text-sm font-medium text-white py-2 px-4 rounded hover:bg-red-600">
              Next
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <router-view :key="$route.fullPath"/>
</template>

<style scoped>

</style>
