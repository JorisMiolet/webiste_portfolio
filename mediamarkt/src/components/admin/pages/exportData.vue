<script>
import imageData from './dummyData.json';
import {PCImage} from "@/image.js";
export default {
  name: 'exportData',
  components: {},
  data(){
    return {
      images: [],
      dummyData: imageData,
      selectedImage: null,
    }
  },
  created() {
    for (let i = 1; i <= 10; i++) {
      const laptopKey = `laptop${i}`;
      if (this.dummyData[laptopKey]) {
        const laptop = this.dummyData[laptopKey];
        this.createOffer(laptop);
      }
    }
   },methods: {
    createOffer(image) {
      image = PCImage.createSampleImage(
          image.ArticleNR,
          image.EAN,
          image.Brand,
          image.Description,
          image.PROCESSOR,
          image.RAM,
          image.STORAGE,
          image.GPU,
      )
      this.images.push(image);
    },
    editImage(pcimage){
      this.selectedImage = pcimage;
      this.toggleView();
    },
    deleteImage(){
      this.images = this.images.filter((image) => image === this.selectedImage);
      this.selectedImage = null;
    },
    resetImage(){
      this.$router.push(this.$route.matched[0].path)
    },
    onSelect(image){
      this.selectedImage = image;
    }
  },
}

</script>

<template>
  <table class="table-auto text-left ml-3">
    <caption class="text-left">
      <button class="bg-red-800 hover:bg-red-500 text-white font-medium mt-2 py-2 px-4 rounded">Export data</button>
      <button class="bg-red-800 hover:bg-red-500 text-white font-medium mt-2 py-2 px-4 rounded">
        <router-link :to="{name: 'createImage'}">Create image</router-link>
      </button>
    </caption>
    <thead class="border-b font-medium dark:border-neutral-500">
    <tr>
      <th scope="col" class="px-6 py-4">#</th>
      <th scope="col" class="px-6 py-4">Article NR</th>
      <th scope="col" class="px-6 py-4">EAN</th>
      <th scope="col" class="px-6 py-4">Brand</th>
      <th scope="col" class="px-6 py-4">Description</th>
      <th scope="col" class="px-6 py-4">Processor</th>
      <th scope="col" class="px-6 py-4">Edit image</th>
    </tr>
    </thead>
    <tbody>
    <tr class="border-b dark:border-neutral-500" v-for="(pcimage, key) in images" :key="pcimage.EAN">
      <td class="whitespace-nowrap px-6 py-4 font-medium">{{ (key+1) }}</td>
      <td class="whitespace-nowrap px-6 py-4">{{ pcimage.ArticleNR }}</td>
      <td class="whitespace-nowrap px-6 py-4">{{ pcimage.EAN }}</td>
      <td class="whitespace-nowrap px-6 py-4">{{ pcimage.PCBrand }}</td>
      <td class="whitespace-nowrap px-6 py-4">{{ pcimage.Description }}</td>
      <td class="whitespace-nowrap px-6 py-4">{{ pcimage.PROCESSOR }}</td>
      <td class="whitespace-nowrap px-6 py-4">
        <button>
          <router-link :to="{name:'editImage', params: {ArticleNR: pcimage.ArticleNR}}" @click="onSelect(pcimage)">edit</router-link>
        </button>
      </td>
    </tr>
    </tbody>
  </table>
  <router-view :key="$route.fullPath"/>
</template>

<style scoped>

</style>