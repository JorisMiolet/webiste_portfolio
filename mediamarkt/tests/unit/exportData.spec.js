import exportData from "@/components/admin/pages/image/exportData.vue";
import { shallowMount } from '@vue/test-utils';


describe('ExportData', () => {

    // Test 1: Mount the component and check if it renders correctly
    it('renders correctly', () => {
        const wrapper = shallowMount(exportData);
        expect(wrapper.element).toMatchSnapshot();
    });

    // Test 2: Test the createOffer method
    it('creates an offer when createOffer method is called', () => {
        // Arrange data
        const wrapper = shallowMount(exportData);
        const image = {
            "ArticleNR": undefined, "Description": undefined, "EAN": "1234567890", "GPU": "NVIDIA GeForce GTX 1650", "PCBrand": undefined, "PROCESSOR": "Intel Core i7", "RAM": "16GB", "STORAGE": "512GB SSD"
        };

        // call function
        wrapper.vm.create_image(image);

        // Assert that the images array is updated with the new image
        expect(wrapper.vm.images).toEqual([image]);
    });

    // Test 3: Test the deleteImage method
    it('deletes an image when deleteImage method is called', async () => {
        const wrapper = shallowMount(exportData);
        const image = {
            "ArticleNR": undefined, "Description": undefined, "EAN": "1234567890", "GPU": "NVIDIA GeForce GTX 1650", "PCBrand": undefined, "PROCESSOR": "Intel Core i7", "RAM": "16GB", "STORAGE": "512GB SSD"
        };
        // Mock the axios delete method
        const axiosDeleteMock = jest.fn(() => Promise.resolve());
        wrapper.vm.axios.post = axiosDeleteMock;

        // Call the deleteImage method
        await wrapper.vm.deleteImage(image, axiosDeleteMock());

        // Assert that the loadAllImages method is called after deletion
        expect(wrapper.vm.axiosDeleteMock).toHaveBeenCalled();
    });

    // Test 4: Test the filterImages method
    it('filters images when filterImages method is called', async () => {
        const wrapper = shallowMount(exportData);
        const image = [{    "ArticleNR": undefined, "Description": undefined, "EAN": "1234567890", "GPU": "NVIDIA GeForce GTX 1650", "PCBrand": undefined, "PROCESSOR": "Intel Core i7", "RAM": "16GB", "STORAGE": "512GB SSD"
        }]
        jest.mock('axios');

        // Mock the axios get method
        const axiosGetMock = jest.fn(() => Promise.resolve({ data: image}));
        wrapper.vm.axios.get = axiosGetMock;

        // Call the deleteImage method
        await wrapper.vm.filterImages();

        // Assert that the loadAllImages method is called after deletion
        expect(wrapper.vm.loadAllImages).toHaveBeenCalled();
    });

    // Test 5: Test the loadAllImages method
    it('loads all images when loadAllImages method is called', async () => {
        // Arrange data
        const wrapper = shallowMount(exportData);
        const image = [{
            "ArticleNR": undefined, "Description": undefined, "EAN": "1234567890", "GPU": "NVIDIA GeForce GTX 1650", "PCBrand": undefined, "PROCESSOR": "Intel Core i7", "RAM": "16GB", "STORAGE": "512GB SSD"
        }]

        // Mock the axios get method
        const axiosGetMock = jest.fn(() => Promise.resolve({ data: image }));
        wrapper.vm.axios.get = axiosGetMock;

        // Call the loadAllImages method
        await wrapper.vm.loadAllImages();

        // Assert that the images array is updated with the loaded data
        expect(wrapper.vm.images).toEqual([{
            "ArticleNR": undefined, "Description": undefined, "EAN": "1234567890", "GPU": "NVIDIA GeForce GTX 1650", "PCBrand": undefined, "PROCESSOR": "Intel Core i7", "RAM": "16GB", "STORAGE": "512GB SSD"
        }]);
    });
});