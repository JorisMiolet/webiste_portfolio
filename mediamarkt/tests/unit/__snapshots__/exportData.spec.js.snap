// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportData renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="p-6"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 mb-6 mt-40 md:mt-0"
    >
      <div
        class="bg-white rounded-md border border-gray-100 p-6 shadow-md shadow-black/5"
      >
        <div
          class="flex justify-between mb-6"
        >
          <div>
            <div
              class="text-2xl font-semibold mb-1"
            >
              0
            </div>
            <div
              class="text-sm font-medium text-gray-600"
            >
              Completed
            </div>
          </div>
        </div>
      </div>
      <div
        class="bg-white rounded-md border border-gray-100 p-6 shadow-md shadow-black/5"
      >
        <div
          class="flex justify-between mb-6"
        >
          <div>
            <div
              class="text-2xl font-semibold mb-1"
            >
              0
            </div>
            <div
              class="text-sm font-medium text-gray-600"
            >
              In progress
            </div>
          </div>
        </div>
      </div>
      <div
        class="bg-white rounded-md border border-gray-100 p-6 shadow-md shadow-black/5"
      >
        <div
          class="flex justify-between mb-6"
        >
          <div>
            <div
              class="text-2xl font-semibold mb-1"
            >
              0
            </div>
            <div
              class="text-sm font-medium text-gray-600"
            >
              Outdated
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mb-6"
    >
      <div
        class="bg-white border border-gray-100 shadow-md shadow-black/5 p-6 rounded-md"
      >
        <div
          class="flex justify-center sm:justify-between mb-4 sm:items-start"
        >
          <div
            class="font-medium"
          >
            Manage images
          </div>
        </div>
        <div
          class="flex flex-col sm:flex-row items-center mb-4 order-tab"
        >
          <button
            class="bg-gray-50 text-sm font-medium text-gray-400 py-2 px-4 rounded-tl-md rounded-bl-md hover:text-gray-600 focus:outline-none active"
            type="button"
          >
            All
          </button>
          <button
            class="text-gray-600 text-sm font-medium text-gray-400 py-2 px-4 hover:text-gray-600 focus:outline-none active"
            type="button"
          >
            Completed
          </button>
          <button
            class="text-gray-600 text-sm font-medium text-gray-400 py-2 px-4 hover:text-gray-600 focus:outline-none active"
            type="button"
          >
            In progress
          </button>
          <button
            class="text-gray-600 text-sm font-medium text-gray-400 py-2 px-4 rounded-tr-md rounded-br-md hover:text-gray-600 focus:outline-none active"
            type="button"
          >
            Outdated
          </button>
          <button
            class="sm:ml-auto bg-red-500 text-sm font-medium text-white py-2 px-4 rounded-tr-md rounded hover:bg-red-600"
            type="button"
          >
            <router-link
              to="[object Object]"
            >
              Create image
            </router-link>
          </button>
        </div>
        <div
          class="flex items-center mb-4"
        >
          <div
            class="relative w-full mr-2"
          >
            <input
              class="py-2 pr-4 pl-10 bg-gray-50 w-full outline-none border border-gray-100 rounded-md text-sm focus:border-blue-500"
              placeholder="Search..."
              type="text"
            />
            <i
              class="ri-search-line absolute top-1/2 left-4 -translate-y-1/2 text-gray-400"
            />
          </div>
        </div>
        <div
          class="overflow-x-auto"
        >
          <table
            class="w-full min-w-[540px]"
            data-page="active"
            data-tab-for="order"
          >
            <thead>
              <tr>
                <th
                  class="text-[12px] uppercase tracking-wide font-medium text-gray-400 py-2 px-4 bg-gray-50 text-left rounded-tl-md rounded-bl-md"
                >
                  Article number
                </th>
                <th
                  class="text-[12px] uppercase tracking-wide font-medium text-gray-400 py-2 px-4 bg-gray-50 text-left"
                >
                  EAN
                </th>
                <th
                  class="text-[12px] uppercase tracking-wide font-medium text-gray-400 py-2 px-4 bg-gray-50 text-left"
                >
                  Brand
                </th>
                <th
                  class="text-[12px] uppercase tracking-wide font-medium text-gray-400 py-2 px-4 bg-gray-50 text-left"
                >
                  Status
                </th>
                <th
                  class="text-[12px] uppercase tracking-wide font-medium text-gray-400 py-2 px-4 bg-gray-50 text-left rounded-tr-md rounded-br-md"
                >
                  User
                </th>
                <th
                  class="text-[12px] uppercase tracking-wide font-medium text-gray-400 py-2 px-4 bg-gray-50 text-left rounded-tr-md rounded-br-md"
                >
                  Actions
                </th>
              </tr>
            </thead>
            <tbody>
              
              
            </tbody>
          </table>
          <div
            class="flex justify-center mt-4"
          >
            <button
              class="disabled bg-red-500 text-sm font-medium text-white py-2 px-4 rounded hover:bg-red-600"
              disabled=""
            >
               Previous 
            </button>
            <div
              class="mx-4"
            >
               Page 1 of 0
            </div>
            <button
              class="bg-red-500 text-sm font-medium text-white py-2 px-4 rounded hover:bg-red-600"
            >
               Next 
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <router-view />
  
</div>
`;
